kernelci/dashboard
# Feat: TreeDetails query and filtering unification

- Refactors TreeDetailsSlowView to be a bit clearer
- Adds build data from TreeDetails to TreeDetailsSlowView, including treeUrl with the same logic as it was from the frontend
- Changes the endpoint used by the frontend to call only TreeDetailsSlowView
- Refactors old endpoint and frontend hook to simply "TreeDetailsView" and "useTreeDetails"

Since the builds data were moved to the backend logic of the old treeDetailsSlowView, all global filters from boots/tests tab now work on builds tab, including the hardware filter. Also, now unknown configs, architectures and compilers are also counted in the build tab (with some masking in the frontend to not clutter the tables)

## How to test
Go to a tree details page with more than one hardware, filter by hardware and then the build tab should change to contain only builds with tests that contained that hardware (such as this [localhost example](http://localhost:5173/tree/b7bfaa761d760e72a969d116517eaa12e404c262?tableFilter=%7B%22buildsTable%22%3A%22all%22%2C%22bootsTable%22%3A%22all%22%2C%22testsTable%22%3A%22all%22%7D&origin=maestro&currentPageTab=global.builds&diffFilter=%7B%22hardware%22%3A%7B%22google%2Cjuniper-sku16%22%3Afalse%2C%22google%2Ckingoftown%22%3Atrue%7D%7D&treeInfo=%7B%22gitBranch%22%3A%22for-kernelci%22%2C%22gitUrl%22%3A%22https%3A%2F%2Fgit.kernel.org%2Fpub%2Fscm%2Flinux%2Fkernel%2Fgit%2Fbroonie%2Fmisc.git%22%2C%22treeName%22%3A%22broonie-misc%22%2C%22commitName%22%3A%22v6.2-rc3%22%2C%22headCommitHash%22%3A%22b7bfaa761d760e72a969d116517eaa12e404c262%22%7D&intervalInDays=7)). Also compare to the [equivalent page on staging](https://staging.dashboard.kernelci.org:9000/tree/b7bfaa761d760e72a969d116517eaa12e404c262?tableFilter=%7B%22buildsTable%22%3A%22all%22%2C%22bootsTable%22%3A%22all%22%2C%22testsTable%22%3A%22all%22%7D&origin=maestro&currentPageTab=global.builds&diffFilter=%7B%22hardware%22%3A%7B%22google%2Cjuniper-sku16%22%3Afalse%2C%22google%2Ckingoftown%22%3Atrue%7D%7D&treeInfo=%7B%22gitBranch%22%3A%22for-kernelci%22%2C%22gitUrl%22%3A%22https%3A%2F%2Fgit.kernel.org%2Fpub%2Fscm%2Flinux%2Fkernel%2Fgit%2Fbroonie%2Fmisc.git%22%2C%22treeName%22%3A%22broonie-misc%22%2C%22commitName%22%3A%22v6.2-rc3%22%2C%22headCommitHash%22%3A%22b7bfaa761d760e72a969d116517eaa12e404c262%22%7D&intervalInDays=7).

Closes #601
